src/meter/MeterState.ts

```diff
@@ -1,9 +1,99 @@
-import {
-  UseProgressProps,
-  useProgressState,
-  ProgressStateReturn,
-} from "../progress";
-
-export type UseMeterProps = UseProgressProps;
-export const useMeterState = useProgressState;
-export type MeterStateReturn = ProgressStateReturn;
+import { valueToPercent } from "@chakra-ui/utils";
+
+export interface UseMeterProps {
+  /**
+   * The `value` of the meter indicator.
+   * If `undefined`/`not valid` the meter bar will be equal to `min`
+   */
+  value?: number;
+  /**
+   * The minimum value of the meter
+   */
+  min?: number;
+  /**
+   * The maximum value of the meter
+   */
+  max?: number;
+  /**
+   * The higher limit of min range
+   */
+  low?: number;
+  /**
+   * The lower limit of max range
+   */
+  high?: number;
+  /**
+   * The lower limit of max range
+   */
+  optimum?: number;
+}
+
+export const useMeterState = (props: UseMeterProps = {}) => {
+  const { min = 0, max = 1 } = props;
+  let {
+    value = 0,
+    low = min,
+    high = max,
+    optimum = getDefaultOptimumValue(low, high),
+  } = props;
+  let status: "safe" | "caution" | "danger" | undefined = undefined;
+
+  value = handleInequalities(value, min, max);
+  low = handleInequalities(low, min, max);
+  high = handleInequalities(high, min, max);
+  optimum = handleInequalities(optimum, min, max);
+
+  if (low >= high) low = high;
+  if (high <= low) high = low;
+
+  if (optimum >= low && optimum <= high) {
+    if (value >= low && value <= high) {
+      status = "safe";
+    } else {
+      status = "caution";
+    }
+  }
+
+  if (optimum >= high && optimum <= max) {
+    if (value >= high && value <= max) {
+      status = "safe";
+    } else if (value < high && value >= low) {
+      status = "caution";
+    } else {
+      status = "danger";
+    }
+  }
+
+  if (optimum >= min && optimum <= low) {
+    if (value >= min && value <= low) {
+      status = "safe";
+    } else if (value > low && value <= high) {
+      status = "caution";
+    } else {
+      status = "danger";
+    }
+  }
```

Comment:
After understandin, see if you can minify this.

Reply:
Hmmmmm :thinking: 

We can extract the common logic tho.

```js

const isInRange = (value: number, min: number, max: number) => {
  return value >= min && value <= max;
};

const calculateStatus = (
  value: number,
  optimum: number,
  min: number,
  max: number,
  outRange: number,
) => {
  let status: TStatus = undefined;
  if (isInRange(optimum, min, max)) {
    if (value >= min && value <= max) {
      status = "safe";
    } else if (value < min && value >= outRange) {
      status = "caution";
    } else {
      status = "danger";
    }
  }

  return status;
};

  // especial case
  if (isInRange(optimum, low, high)) {
    if (value >= low && value <= high) {
      status = "safe";
    } else {
      status = "caution";
    }
  }

  status = calculateStatus(value, optimum, high, max, low);
  status = calculateStatus(value, optimum, min, low, high);
```

Reply:
```js
if (isInRange(optimum, min, max)) {
    if (value >= min && value <= max) {
      status = "safe";
    } else if (value < min && value >= outRange) {
      status = "caution";
    } else {
      status = "danger";
    }
  }
```

else if changes for two cases, I guess you did not notice. I have made changes and minified it.